### DISCUSSION TITLE
How to clear cached remote files?

### DISCUSSION THREAD

**Discussion by vlnst** [misskey-dev/misskey#13575] [Q&A]
![image](https://github.com/misskey-dev/misskey/assets/77411099/9b043553-e137-44b2-9b54-f16d5cde23cd)
After enabling this I thought I can disable it and clear remote files, but it turns out that's not the case (I think this string should be edited because it's misleading admins). 
Now I have over 100GB of random images that might be considered illegal in my country.
I've tried to solve this by using data from DB, but it didn't clean all remote files, just a few.

I've found some issues about this problem but none of them were resolved (#9237, #11493, #12107)

Is there any way to clear this cache?

**Comment by tassoman** [comment]
I'm thinking about writing a pruning script, who deletes remote posts and their attachments by API calls. :thinking:

**Comment by vlnst** [comment]
Well, I've removed this line from postgres.ts https://github.com/misskey-dev/misskey/blob/8d23122fd664564dc069ca8e8e337f4d4a1727fc/packages/backend/src/postgres.ts#L215
but it didn't help much. Now It says it cleared the cache in logs but in reality it is not

**Reply by tassoman** [reply]
Well, I discovered since 2024.8.0, there's a trash bin icon 🗑️ in Drive Admin page. The trash icon deletes remote items local cache.
As far I know, there isn't any cron or recurrence activity set. Must be clicked sometimes, maybe each week or each month.
I'm not really sure it will work deleting remote file caching. Although API call is this: `admin/drive/clean-remote-files`.

**Reply by vlnst** [reply]
Yeah, I tried clicking the icon in admin interface and using API with curl.

These are the logs:

```
INFO 1	[queue clean-remote-files]	Deleting cached remote files...
```
```
query is slow: SELECT "MiDriveFile"."id" AS "MiDriveFile_id", "MiDriveFile"."userId" AS "MiDriveFile_userId", "MiDriveFile"."userHost" AS "MiDriveFile_userHost", "MiDriveFile"."md5" AS "MiDriveFile_md5", "MiDriveFile"."name" AS "MiDriveFile_name", "MiDriveFile"."type" AS "MiDriveFile_type", "MiDriveFile"."size" AS "MiDriveFile_size", "MiDriveFile"."comment" AS "MiDriveFile_comment", "MiDriveFile"."blurhash" AS "MiDriveFile_blurhash", "MiDriveFile"."properties" AS "MiDriveFile_properties", "MiDriveFile"."storedInternal" AS "MiDriveFile_storedInternal", "MiDriveFile"."url" AS "MiDriveFile_url", "MiDriveFile"."thumbnailUrl" AS "MiDriveFile_thumbnailUrl", "MiDriveFile"."webpublicUrl" AS "MiDriveFile_webpublicUrl", "MiDriveFile"."webpublicType" AS "MiDriveFile_webpublicType", "MiDriveFile"."accessKey" AS "MiDriveFile_accessKey", "MiDriveFile"."thumbnailAccessKey" AS "MiDriveFile_thumbnailAccessKey", "MiDriveFile"."webpublicAccessKey" AS "MiDriveFile_webpublicAccessKey", "MiDriveFile"."uri" AS "MiDriveFile_uri", "MiDriveFile"."src" AS "MiDriveFile_src", "MiDriveFile"."folderId" AS "MiDriveFile_folderId", "MiDriveFile"."isSensitive" AS "MiDriveFile_isSensitive", "MiDriveFile"."maybeSensitive" AS "MiDriveFile_maybeSensitive", "MiDriveFile"."maybePorn" AS "MiDriveFile_maybePorn", "MiDriveFile"."isLink" AS "MiDriveFile_isLink", "MiDriveFile"."requestHeaders" AS "MiDriveFile_requestHeaders", "MiDriveFile"."requestIp" AS "MiDriveFile_requestIp" FROM "drive_file" "MiDriveFile" WHERE ((NOT("MiDriveFile"."userHost" IS NULL)) AND ("MiDriveFile"."isLink" = $1)) ORDER BY "MiDriveFile"."id" ASC LIMIT 8 -- PARAMETERS: [false]
execution time: 27049
DONE 1	[queue clean-remote-files]	All cached remote files has been deleted.
```

It says it went successfully but it didn't remove the files

**Reply by vlnst** [reply]
And yeah, It is can easily be made automatic by using cron with doing POST requests to the API. The problem is the cache doesn't get cleared

**Reply by tassoman** [reply]
If you have  evidence of this, maybe you can file an issue.
I guess empty cache means just file removal, locally in the server. if the related note gets fetched or shown in timeline, that file gets recreated instantly.
